src/vcpkg-test/util-tests.cpp

```diff
@@ -0,0 +1,33 @@
+#include <catch2/catch.hpp>
+
+#include <vcpkg/base/util.h>
+
+namespace Util = vcpkg::Util;
+
+TEST_CASE ("find_nth", "[util]")

```

Comment:
Please add a test for "empty" to protect against off-by-one errors to these. (I think I remember an unprotected base()-1 in the product code?)